### PR TITLE
Use os.path.normpath() for include paths

### PR DESCRIPTION
**Pull request by ClausKlein**
This make relative pathes shorter an too give a chance to
de-duplicate -isystem flags just like -I flags.

Fix common test case 203 for OSX build host too.

Related to https://github.com/mesonbuild/meson/issues/2155

### CODE REVIEW

**Comment by ClausKlein** [comment]
see too the https://github.com/ninja-build/ninja/issues/1251#issuecomment-598684012

File: mesonbuild/mesonlib.py [hunk 1]
```diff
@@ -270,8 +270,10 @@ def from_absolute_file(fname: str) -> 'File':
     def rel_to_builddir(self, build_to_src: str) -> str:
         if self.is_built:
             return self.relative_name()
+            #NO! return os.path.realpath(self.relative_name())
         else:
             return os.path.join(build_to_src, self.subdir, self.fname)
+            #NO! return os.path.realpath(os.path.join(build_to_src, self.subdir, self.fname))
```
Comment (jpakkane) [inline mesonbuild/mesonlib.py]: What are these comments for? I don't understand what they try to say?

File: test cases/common/203 function attributes/meson.build [hunk 2]
```diff
@@ -65,6 +64,7 @@ endif
 if host_machine.system() != 'darwin'
   attributes += 'alias'
   attributes += 'visibility'
+  attributes += 'alloc_size'
```
Comment (jpakkane) [inline test cases/common/203 function attributes/meson.build]: This seems like an unrelated change, it should be in its own MR.

File: mesonbuild/backend/ninjabackend.py [hunk 3]
```diff
@@ -2013,7 +2013,8 @@ def get_custom_target_dir_include_args(self, target, compiler):
             # own target build dir.
             if not isinstance(i, (build.CustomTarget, build.CustomTargetIndex)):
                 continue
-            idir = self.get_target_dir(i)
+            # NOTE: if we want to use os.path.realpath() for include path only! CK
+            idir = os.path.normpath(self.get_target_dir(i))
```
Comment (jpakkane) [inline mesonbuild/backend/ninjabackend.py]: The comment does not make sense. Also, please don't add initials or any such tags to the code. They are just noise, that's what `git blame` is for. The same goes for the other two comments below

It would probably also be better to make `get_target_dir` return normalised paths. This way it fixes the issue for all call sites.

File: mesonbuild/backend/ninjabackend.py [hunk 4]
```diff
@@ -2013,7 +2013,8 @@ def get_custom_target_dir_include_args(self, target, compiler):
             # own target build dir.
             if not isinstance(i, (build.CustomTarget, build.CustomTargetIndex)):
                 continue
-            idir = self.get_target_dir(i)
+            # NOTE: if we want to use os.path.realpath() for include path only! CK
```
Comment (dcbaker) [inline mesonbuild/backend/ninjabackend.py]: I don't understand what this comment even means, (I do know what `os.path.realpath` does). Also, please don't sign your comments, that's what `git blame` is for.

File: test cases/common/203 function attributes/meson.build (hunk 2 above)
Comment (ClausKlein) [reply test cases/common/203 function attributes/meson.build]: without that, i can't run the tests

File: mesonbuild/mesonlib.py (hunk 1 above)
Comment (ClausKlein) [reply mesonbuild/mesonlib.py]: My intension was to use realpath() , but there is still a ninja bug.
see https://github.com/ninja-build/ninja/issues/1251#issuecomment-599541385

File: mesonbuild/backend/ninjabackend.py (hunk 3 above)
Comment (ClausKlein) [reply mesonbuild/backend/ninjabackend.py]: see https://github.com/ninja-build/ninja/issues/1251

File: mesonbuild/backend/ninjabackend.py (hunk 4 above)
Comment (ClausKlein) [reply mesonbuild/backend/ninjabackend.py]: done

File: mesonbuild/backend/ninjabackend.py (hunk 3 above)
Comment (ClausKlein) [reply mesonbuild/backend/ninjabackend.py]: done

File: mesonbuild/mesonlib.py (hunk 1 above)
Comment (ClausKlein) [reply mesonbuild/mesonlib.py]: done

**Comment by jpakkane** [comment]
Thanks.
